src/OpenApi/src/Comparers/OpenApiSchemaComparer.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal sealed class OpenApiSchemaComparer : IEqualityComparer<OpenApiSchema>
```

Comment:
Are we pushing to get this implemented upstream? Is https://github.com/microsoft/OpenAPI.NET/issues/537 what's currently tracking it?

Since we don't control this type, and there's a ton of properties we need to compare, it might be a good idea to add a test similar to `CloneSslOptionsClonesAllProperties` which uses reflection to verify that no public properties have been added that are not cloned. This would presumably fail if a dependency update if a public property gets added without a reaction.

https://github.com/dotnet/aspnetcore/blob/42925a4cebf501fd00aae1a1e899a969b4a2bd9a/src/Servers/Kestrel/Core/test/SniOptionsSelectorTests.cs#L806-L809

Reply:
> Are we pushing to get this implemented upstream? Is https://github.com/microsoft/OpenAPI.NET/issues/537 what's currently tracking it?

Yep, I've brought this up with folks on the OpenAPI.NET side. The equality implementations are useful to have at the lower level.

> Since we don't control this type, and there's a ton of properties we need to compare, it might be a good idea to add a test similar to CloneSslOptionsClonesAllProperties which uses reflection to verify that no public properties have been added that are not cloned.

Oooh...good idea. I'll take a look at this test.